src/GitHub.Api/Installer/GitInstaller.cs

```diff
@@ -11,206 +12,183 @@ class GitInstaller
+        public GitInstallationState SetupGitIfNeeded()
         {
             //Logger.Trace("SetupGitIfNeeded");
-            GitInstallationState installationState = new GitInstallationState();
-            installationTask = new FuncTask<GitInstallationState, GitInstallationState>(cancellationToken, (success, path) => path)
-                { Name = "Git Installation - Complete" };
-            installationTask.OnStart += thisTask => thisTask.UpdateProgress(0, 100);
-            installationTask.OnEnd += (thisTask, result, success, exception) => thisTask.UpdateProgress(100, 100);
+            var state = new GitInstallationState();
 
-            ITask<GitInstallationState> startTask = null;
-            if (!environment.IsWindows)
+            if (systemSettings != null)
             {
-                var findTask = new FindExecTask("git", cancellationToken)
-                    .Configure(processManager, dontSetupGit: true)
-                    .Catch(e => true);
-                findTask.OnEnd += (thisTask, path, success, exception) =>
+                NPath? gitExecutablePath = systemSettings.Get(Constants.GitInstallPathKey)?.ToNPath();
+                if (gitExecutablePath.HasValue && gitExecutablePath.Value.FileExists()) // we have a git path
                 {
-                    // we should doublecheck that system git is usable here
-                    installationState.GitIsValid = success;
-                    if (success)
-                    {
-                        installationState.GitExecutablePath = path;
-                        installationState.GitInstallationPath = path.Resolve().Parent.Parent;
-                    }
-                };
-                findTask.Then(new FindExecTask("git-lfs", cancellationToken)
-                    .Configure(processManager, dontSetupGit: true))
-                    .Catch(e => true);
-                findTask.OnEnd += (thisTask, path, success, exception) =>
-                {
-                    installationState.GitLfsIsValid = success;
-                    if (success)
-                    {
-                        // we should doublecheck that system git is usable here
-                        installationState.GitLfsExecutablePath = path;
-                        installationState.GitLfsInstallationPath = path.Resolve().Parent.Parent;
-                    }
-                };
-                startTask = findTask.Then(s => installationState);
-            }
-            else
-            {
-                startTask = new FuncTask<GitInstallationState>(cancellationToken, () =>
-                    {
-                        return VerifyPortableGitInstallation();
-                    })
-                { Name = "Git Installation - Verify" };
+                    Logger.Trace("Using git install path from settings: {0}", gitExecutablePath);
+                    state.GitExecutablePath = gitExecutablePath.Value;
+                    state.GitIsValid = true;
```

Comment:
If someone provides us with a custom git. Shouldn't we search for Git LFS locally on the system here. Currently it seems that if someone gives us a copy of git we are still going to use our own Git LFS.